### PR TITLE
Add k8s 1.25 to kubeadm tests

### PR DESCRIPTION
**Pull request by krnowak**
Also comes with a script that get the checksums.

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

### CODE REVIEW

File: kola/tests/kubeadm/checksums.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+tuples=(
+    "KubeadmSum^https://storage.googleapis.com/kubernetes-release/release/%K8S_VER%/bin/linux/%ARCH%/kubeadm"
+    "KubeletSum^https://storage.googleapis.com/kubernetes-release/release/%K8S_VER%/bin/linux/%ARCH%/kubelet"
+    "CRIctlSum^https://github.com/kubernetes-sigs/cri-tools/releases/download/%CRICTL_VER%/crictl-%CRICTL_VER%-linux-%ARCH%.tar.gz"
+    "CNISum^https://github.com/containernetworking/plugins/releases/download/%CNI_VER%/cni-plugins-linux-%ARCH%-%CNI_VER%.tgz"
+    "KubectlSum^https://storage.googleapis.com/kubernetes-release/release/%K8S_VER%/bin/linux/%ARCH%/kubectl"
+)
+
+this_dir=$(dirname $0)
```
Comment (pothos) [inline kola/tests/kubeadm/checksums.sh:27]:
```suggestion
this_dir=$(dirname "$0")
```
Shouldn't be a problem normally but who knows ;)

Comment (krnowak) [reply kola/tests/kubeadm/checksums.sh:27]: Not planning to run it in `C:\Program Files (x86)\…` any time soon, but right you are. :)

Comment (krnowak) [reply kola/tests/kubeadm/checksums.sh:27]: Will fix it after the docker image is built, so I can run the tests on Jenkins.

Comment (krnowak) [reply kola/tests/kubeadm/checksums.sh]: Nevermind, fixed now. Also did a rebase, so the new docker image will include the extra tests for cilium.

File: kola/tests/kubeadm/checksums.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -euo pipefail
+
+fail() {
+    IFS=' ' echo "$*" >&2
+    exit 1
+}
+
```
Comment (pothos) [inline kola/tests/kubeadm/checksums.sh:9]: I think usage instructions would be good here when `-h` is passed

Comment (pothos) [reply kola/tests/kubeadm/checksums.sh:9]: and how to add a new entry (e.g., first copy the map entry and set the new version but leave the wrong checksums, then run it and update the checksums)

Comment (krnowak) [reply kola/tests/kubeadm/checksums.sh:9]: Sure, will do.

**Comment by krnowak** [comment]
Running the tests:
- http://jenkins.infra.kinvolk.io:8080/job/container/job/test/2060 - amd64 qemu
- http://jenkins.infra.kinvolk.io:8080/job/container/job/test/2064 - arm64 qemu_uefi
- http://jenkins.infra.kinvolk.io:8080/job/container/job/test/2062 - amd64 azure
- http://jenkins.infra.kinvolk.io:8080/job/container/job/test/2063 - arm64 azure

File: kola/tests/kubeadm/checksums.sh (hunk 2 above)
Comment (krnowak) [reply kola/tests/kubeadm/checksums.sh:9]: Done.

Comment (pothos) [reply kola/tests/kubeadm/checksums.sh:9]: Thanks

**Comment by krnowak** [comment]
Running the tests again (testing my test dispatcher job at the same time):
- amd64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/2/cldsv/
- arm64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/3/cldsv/

**Comment by krnowak** [comment]
It looks like it will need more investigation:

```
[2022-08-25T13:38:22.437Z] --- FAIL: kubeadm.v1.25.0.cilium.base (1085.71s)
[2022-08-25T13:38:22.437Z]         kubeadm.go:248: unable to setup cluster: unable to create master node: Future#WaitForCompletion: context has been cancelled: StatusCode=200 -- Original Error: context deadline exceeded
```

File: kola/tests/kubeadm/kubeadm.go [hunk 3]
```diff
@@ -71,6 +71,37 @@ var (
 	// testConfig holds params for various kubernetes releases
 	// and the nested params are used to render script templates
 	testConfig = map[string]map[string]interface{}{
+		"v1.25.0": map[string]interface{}{
+			// from https://github.com/flannel-io/flannel/releases
+			"FlannelVersion": "v0.19.1",
+			// from https://github.com/cilium/cilium/releases
+			"CiliumVersion": "1.12.1",
+			// from https://raw.githubusercontent.com/cilium/cilium/master/stable.txt
+			"CiliumCLIVersion": "v1.12.1",
```
Comment (tormath1) [inline kola/tests/kubeadm/kubeadm.go:80]:
```suggestion
			"CiliumCLIVersion": "v0.12.1",
```

This is the latest release: https://github.com/cilium/cilium-cli/releases/tag/v0.12.1 for the CLI.

Comment (tormath1) [inline kola/tests/kubeadm/kubeadm.go:80]:
```suggestion
			// from https://github.com/cilium/cilium-cli/releases/
			"CiliumCLIVersion": "v0.12.1",
```
This is the latest release: https://github.com/cilium/cilium-cli/releases/tag/v0.12.1 for the CLI.

Comment (krnowak) [reply kola/tests/kubeadm/kubeadm.go]: Thanks for spotting it. Fixed.

**Comment by krnowak** [comment]
- amd64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/6/cldsv
- arm64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/7/cldsv

**Comment by krnowak** [comment]
Now it fails somewhere inside `cilium install` with a message `cilium version not found: open cilium-1.12.1.tgz: file does not exist`. This line actually comes from this place: https://github.com/cilium/cilium-cli/blob/master/internal/helm/helm.go#L186-L194

**Comment by tormath1** [comment]
> Now it fails somewhere inside `cilium install` with a message `cilium version not found: open cilium-1.12.1.tgz: file does not exist`. This line actually comes from this place: https://github.com/cilium/cilium-cli/blob/master/internal/helm/helm.go#L186-L194

It looks like Cilium CLI now uses Helm charts to generate Cilium manifest. Helm charts are versioned in the vendors, and the version 1.12.1 has been added in the following commit: https://github.com/cilium/cilium-cli/commit/4216e69ac7fe699ee147948db8cd7d7f22b424b3 not yet released.

**Comment by krnowak** [comment]
Bumped cilium cli to 0.12.2 (released 4 days ago), so it should now support cilium 1.12.1. Let's see now.

**Comment by krnowak** [comment]
Running the tests again:
- amd64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/8/cldsv
- arm64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/9/cldsv

**Comment by tormath1** [comment]
while trying this branch:
```
  Normal   Pulled     107s (x5 over 3m17s)  kubelet            Container image "quay.io/cilium/cilium:v1.12.1@sha256:ea2db1ee21b88127b5c18a96ad155c25485d0815a667ef77c2b7c7f31cab601b" already present on machine
  Normal   Created    107s (x5 over 3m17s)  kubelet            Created container mount-cgroup
  Warning  Failed     107s (x5 over 3m17s)  kubelet            Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: write /proc/self/attr/keycreate: invalid argument: unknown
  Warning  BackOff    93s (x10 over 3m16s)  kubelet            Back-off restarting failed container
```

that sounds like a SELinux issue but got the same result with permissive mode.

**Comment by krnowak** [comment]
Running the tests again:
- amd64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/12/cldsv
- arm64: http://jenkins.infra.kinvolk.io:8080/job/container/job/test_dispatcher/13/cldsv

**Review by tormath1 — APPROVED** [review]
CI looks good. Feel free to merge once rebased on `flatcar-master`. :tada: Thanks for the script :)

**Pull request merged**